.github/workflows/dart_ci.yml

```diff
@@ -49,6 +50,11 @@ jobs:
           if [[ "$DART_VERSION" =~ ^3 ]]; then
             ./tool/delete_dart_2_only_files.sh
           fi
+      - name: Switch to React 17 Test HTML
+        if: ${{ matrix.react == 17 }}
+        run: |
+          mv test/templates/html_template.html test/templates/html_template-old.html
+          mv test/templates/html_template-react17.html test/templates/html_template.html
```

Comment:
Oh one more thing I thought of: could we add some automated validation that this swap works, and tests are running against the React version we expect?

E.g., a test with some tags like described here: https://github.com/Workiva/react_testing_library/blob/dac2b64da94f6d1e2486a5a0b98caadaa40d72b6/test/unit/react_version_test.dart#L14-L23